server/controllers/collection.controller/addProjectToCollection.js

```diff
@@ -69,9 +62,19 @@ export default function addProjectToCollection(req, res) {
     ]);
   }
 
-  return Promise.all([collectionPromise, projectPromise])
-    .then(updateCollection)
-    .then(populateReferences)
-    .then(sendSuccess)
-    .catch(sendFailure);
+  try {
+    const collection = await Collection.findById(collectionId).populate(
+      'items.project',
+      '_id'
+    );
+    const project = await Project.findById(projectId);
```

Comment:
There's actually a slight performance benefit to the `Promise.all` in the previous code because it allows the two "findById" functions to be executed at the same time instead of one and then the other.